lib/src/fsmonitor.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2020 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Interfaces with a filesystem monitor tool to efficiently query for
+//! filesystem updates, without having to crawl the entire working copy. This is
+//! particularly useful for large working copies, or for working copies for
+//! which it's expensive to materialize files, such those backed by a network or
+//! virtualized filesystem.
+
+#![warn(missing_docs)]
```

Comment:
If we think it's a good idea to enforce documentation (seems like it to me), we should move this to the top of the crate, and maybe add temporary `allow(missing_docs)`. That's obviously out of scope for this PR.